// patternfly-react/dist/js/components/Cards/UtilizationTrendCard/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UtilizationCardDetailsLine2 = exports.UtilizationCardDetailsLine1 = exports.UtilizationCardDetailsDesc = exports.UtilizationCardDetailsCount = exports.UtilizationCardDetails = exports.UtilizationCard = undefined;

var _UtilizationCard = require('./UtilizationCard');

var _UtilizationCard2 = _interopRequireDefault(_UtilizationCard);

var _UtilizationCardDetails = require('./UtilizationCardDetails');

var _UtilizationCardDetails2 = _interopRequireDefault(_UtilizationCardDetails);

var _UtilizationCardDetailsCount = require('./UtilizationCardDetailsCount');

var _UtilizationCardDetailsCount2 = _interopRequireDefault(_UtilizationCardDetailsCount);

var _UtilizationCardDetailsDesc = require('./UtilizationCardDetailsDesc');

var _UtilizationCardDetailsDesc2 = _interopRequireDefault(_UtilizationCardDetailsDesc);

var _UtilizationCardDetailsLine = require('./UtilizationCardDetailsLine1');

var _UtilizationCardDetailsLine2 = _interopRequireDefault(_UtilizationCardDetailsLine);

var _UtilizationCardDetailsLine3 = require('./UtilizationCardDetailsLine2');

var _UtilizationCardDetailsLine4 = _interopRequireDefault(_UtilizationCardDetailsLine3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

_UtilizationCard2.default.Details = _UtilizationCardDetails2.default;
_UtilizationCard2.default.DetailsCount = _UtilizationCardDetailsCount2.default;
_UtilizationCard2.default.DetailsDesc = _UtilizationCardDetailsDesc2.default;
_UtilizationCard2.default.DetailsLine1 = _UtilizationCardDetailsLine2.default;
_UtilizationCard2.default.DetailsLine2 = _UtilizationCardDetailsLine4.default;

exports.UtilizationCard = _UtilizationCard2.default;
exports.UtilizationCardDetails = _UtilizationCardDetails2.default;
exports.UtilizationCardDetailsCount = _UtilizationCardDetailsCount2.default;
exports.UtilizationCardDetailsDesc = _UtilizationCardDetailsDesc2.default;
exports.UtilizationCardDetailsLine1 = _UtilizationCardDetailsLine2.default;
exports.UtilizationCardDetailsLine2 = _UtilizationCardDetailsLine4.default;
