// patternfly-react/dist/js/components/LoginPage/assets/img/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _dropboxLogo = require('patternfly/dist/img/dropbox-logo.svg');

var _dropboxLogo2 = _interopRequireDefault(_dropboxLogo);

var _facebookLogo = require('patternfly/dist/img/facebook-logo.svg');

var _facebookLogo2 = _interopRequireDefault(_facebookLogo);

var _fedoraLogo = require('patternfly/dist/img/fedora-logo.png');

var _fedoraLogo2 = _interopRequireDefault(_fedoraLogo);

var _gitLogo = require('patternfly/dist/img/git-logo.svg');

var _gitLogo2 = _interopRequireDefault(_gitLogo);

var _githubLogo = require('patternfly/dist/img/github-logo.svg');

var _githubLogo2 = _interopRequireDefault(_githubLogo);

var _googleLogo = require('patternfly/dist/img/google-logo.svg');

var _googleLogo2 = _interopRequireDefault(_googleLogo);

var _instagramLogo = require('patternfly/dist/img/instagram-logo.png');

var _instagramLogo2 = _interopRequireDefault(_instagramLogo);

var _linkedinLogo = require('patternfly/dist/img/linkedin-logo.svg');

var _linkedinLogo2 = _interopRequireDefault(_linkedinLogo);

var _openIdLogo = require('patternfly/dist/img/open-id-logo.svg');

var _openIdLogo2 = _interopRequireDefault(_openIdLogo);

var _skypeLogo = require('patternfly/dist/img/skype-logo.svg');

var _skypeLogo2 = _interopRequireDefault(_skypeLogo);

var _stackExchangeLogo = require('patternfly/dist/img/stack-exchange-logo.svg');

var _stackExchangeLogo2 = _interopRequireDefault(_stackExchangeLogo);

var _twitterLogo = require('patternfly/dist/img/twitter-logo.svg');

var _twitterLogo2 = _interopRequireDefault(_twitterLogo);

var _bgLogin = require('patternfly/dist/img/bg-login.jpg');

var _bgLogin2 = _interopRequireDefault(_bgLogin);

var _Logo_Horizontal_Reversed = require('patternfly/dist/img/Logo_Horizontal_Reversed.svg');

var _Logo_Horizontal_Reversed2 = _interopRequireDefault(_Logo_Horizontal_Reversed);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  dropbox: _dropboxLogo2.default,
  facebook: _facebookLogo2.default,
  fedora: _fedoraLogo2.default,
  git: _gitLogo2.default,
  github: _githubLogo2.default,
  google: _googleLogo2.default,
  instagram: _instagramLogo2.default,
  linkedin: _linkedinLogo2.default,
  openID: _openIdLogo2.default,
  skype: _skypeLogo2.default,
  stackExchange: _stackExchangeLogo2.default,
  twitter: _twitterLogo2.default,
  background: _bgLogin2.default,
  brand: _Logo_Horizontal_Reversed2.default
};
